app/components/filter-distance-from-point.js

```diff
@@ -0,0 +1,58 @@
+import Component from '@ember/component';
+// import { argument } from '@ember-decorators/argument';
+import { action, computed } from '@ember-decorators/object';
+import { generateCircleFromFeet } from 'labs-zap-search/helpers/generate-circle-from-feet';
+
+// it takes a point and radius, computes a geojson circle
+// and handles clicks that query mapbox-gl for intersections
+// triggers optional events
+export default class FilterDistanceFromPoint extends Component {
+  // @argument
+  map;
+
+  // @argument
+  pointGeometry;
+
+  // @argument
+  radius;
+
+  // @argument
+  onRadiusFilterClick = () => {}
+
+  // @argument
+  pointLayerId = 'project-centroids-circle';
+
+  // @argument
+  shouldQueryFullMap = false;
+
+  // geojson
+  @computed('pointGeometry', 'radius')
+  get circleFromRadius() {
+    const { pointGeometry, radius } = this;
+
+    return generateCircleFromFeet([pointGeometry, radius]);
+  }
+
+  // queries relevant layer for intersecting feature and sends it
+  // to the click action. conditionally allows for clicking
+  // anywhere on the map
+  @action
+  handleClick(e) {
+    const { target: map } = e;
+    const { point } = e;
+    const [feature] = map.queryRenderedFeatures(
+      point,
+      { layers: [this.pointLayerId] },
+    );
+
+    if (feature) {
```

Comment:
I console logged `if (feature)` vs. `else if (this.shouldQueryFullMap)` and I couldn't find a case when `if (feature)` was true. I'm assuming this is checking whether a feature point for the radius already exists on the map, so not sure why this wouldn't be true if I had made multiple points. Could you explain a little more what this does/is checking for?

Reply:
@trbmcginnis did you try logged it when the filter is off and you click on a point?